crates/ra_ide_api/src/references/search_scope.rs

```diff
@@ -0,0 +1,61 @@
+use hir::{DefWithBody, HasSource, ModuleSource};
+use ra_db::{FileId, SourceDatabase};
+use ra_syntax::{AstNode, TextRange};
+
+use crate::db::RootDatabase;
+
+use super::{NameDefinition, NameKind};
+
+pub(crate) struct SearchScope {
+    pub scope: Vec<(FileId, Option<TextRange>)>,
```

Comment:
I've found that making non-trivial fields public at the start with "nah, I'll refactor later" leads to rather painful refactorings. Let's expose this as an `impl Iterator<>`